src/chains/signals.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import os
+from urllib.parse import urljoin
+
+import requests
+from django.db.models.signals import post_delete, post_save
+from django.dispatch import receiver
+
+from .models import Chain
+
+logger = logging.getLogger(__name__)
+
+
+def setup_session():
+    session = requests.Session()
+    adapter = requests.adapters.HTTPAdapter(max_retries=3)
+    session.mount("http://", adapter)
+    session.mount("https://", adapter)
+    return session
+
+
+session = setup_session()
```

Comment:
Instead of having a global variable you can add the https://docs.python.org/3/library/functools.html#functools.cache annotation to the function, yous FYI, but I think it's good this way too 😉